### PR TITLE
fix mspVtx issue

### PR DESCRIPTION
**Pull request by ligenxxxx**
[https://github.com/hd-zero/hdzero-vtx/issues/14](https://github.com/hd-zero/hdzero-vtx/issues/14)
@rquellet 
- After exiting 0mW, a brief rainbow is displayed when arming.
- When mspVtx is detected, it is initialized when FC configures mspVtx, and supports 0mW startup.As an additional problem, without SA and mspVtx, rf initialization is delayed by 10 seconds.

### CODE REVIEW

**Comment by ligenxxxx** [comment]
[vtx_freestyle.zip](https://github.com/hd-zero/hdzero-vtx/files/9204587/vtx_freestyle.zip)
[vtx_whoop_lite.zip](https://github.com/hd-zero/hdzero-vtx/files/9204588/vtx_whoop_lite.zip)
[vtx_whoop.zip](https://github.com/hd-zero/hdzero-vtx/files/9204589/vtx_whoop.zip)
[vtx_race.zip](https://github.com/hd-zero/hdzero-vtx/files/9204590/vtx_race.zip)

**Comment by rquellet** [comment]
I have tested this on race v2 and whoop vtx's. It doesn't work properly on either one. For whoop vtx, the video never turns on (this is with the proper betaflight 4.4 firmware for msp vtx. For race v2 using smart audio, the drone starts up on r1 always and I have to do a few things to change the channel to r3 for my race.

**Comment by ligenxxxx** [comment]
[vtx_race.zip](https://github.com/hd-zero/hdzero-vtx/files/9231808/vtx_race.zip)

[vtx_whoop.zip](https://github.com/hd-zero/hdzero-vtx/files/9231809/vtx_whoop.zip)

[vtx_whoop_lite.zip](https://github.com/hd-zero/hdzero-vtx/files/9231810/vtx_whoop_lite.zip)

[vtx_freestyle.zip](https://github.com/hd-zero/hdzero-vtx/files/9231811/vtx_freestyle.zip)

Found a bug, when VTX powers up on channel X, VTX can modify channel to Y via mspVtx, but cannot go back to channel X. It has been reported to betaflight and needs to be confirmed with betaflight. **https://github.com/betaflight/betaflight/pull/11705**

File: src/common.h [hunk 1]
```diff
@@ -35,19 +35,26 @@
...
 //#define _DEBUG_SA
 //#define _DEBUG_DISPLAYPORT
 #endif
 
 #define Raceband
 #define USE_EFUSE
-#define USE_MSP
+
+#ifndef _RF_CALIB
+//#define USE_MSP
```
Comment (bkleiner) [inline src/common.h:47]: i think this one should be _active_ by default.

File: src/common.h [hunk 2]
```diff
@@ -35,19 +35,26 @@
...
+#endif
 
 #ifndef VTX_S
 #define USE_SMARTAUDIO
 #endif
+
+#if defined(VTX_L) || defined(VTX_WL)
+#define USE_TEMPERATURA_SENSOR
```
Comment (bkleiner) [inline src/common.h:55]: typo, should be `USE_TEMPERATURE_SENSOR`

**Comment by bkleiner** [comment]
Sorry about the comotion i seem to have caused with my PRs.

i have taken the liberty of creating a rebased version of these changes over here
https://github.com/bkleiner/hdzero-vtx/tree/mspvtx-fix

**Comment by rquellet** [comment]
> [vtx_race.zip](https://github.com/hd-zero/hdzero-vtx/files/9231808/vtx_race.zip)
> 
> [vtx_whoop.zip](https://github.com/hd-zero/hdzero-vtx/files/9231809/vtx_whoop.zip)
> 
> [vtx_whoop_lite.zip](https://github.com/hd-zero/hdzero-vtx/files/9231810/vtx_whoop_lite.zip)
> 
> [vtx_freestyle.zip](https://github.com/hd-zero/hdzero-vtx/files/9231811/vtx_freestyle.zip)
> 
> * Bug fix: The state machine is wrong, causing vtx to not continue to receive MSP data after configuring mspVTX once.
> * Known issue, when VTX powers up on channel X, VTX can modify channel to Y via mspVtx, but cannot go back to channel X. It has been reported to betaflight and needs to be confirmed with betaflight. **[VTX Device over MSP betaflight/betaflight#11705](https://github.com/betaflight/betaflight/pull/11705)**

I tested this on the whoop vtx today with betaflight 4.4 mspVTX build and it worked great! I can start the vtx at 0mW and then use a switch on my radio to change the vtx to a different power level. I still need to test on my race v2 vtx

**Pull request merged**